### ISSUE TITLE
Testing Excel to RDF to RVA publication method (FSDF/vocab-maintenance.md)

### ISSUE THREAD

**Issue by LongJenny** [GeoscienceAustralia/FSDF#5]
@dr-shorthair

Hi Simon,

Thanks for preparing vocab-maintenance.md, outlining the process of publishing in RVA via an Excel to RDF conversion.  I found it easy to follow (even for someone new to data/vocabs like me) and well laid-out.

"FSDF Themes" has been published in RVA using this method.  Can you review the published version and let me know if it all looks OK to you.  Owner is Geoscience Australia - ICSM (GA - ICSM); I requested ARDC include you on this account, but if you have any trouble accessing please let me know.

Issues I've noted, and would appreciate your thoughts on, are:

1. Hyphens and apostrophes in Excel text are not being correctly converted to RDF, and are appearing as clusters of other characters (e.g. see elevation and depth);
2. We need to add a process step to vocab-maintenance.md, outlining how to create an RDF file for import into RVA from the webpage generated by SKOS Play! The method I used was to copy and paste text to notepad++ and save as a Turtle file, but I’m not sure if this was what was intended; 
3. None of the vocabulary metadata came across (see rows 6-15 in "basic example" from vocab-maintenance.md) - is that what you expected?  I'm now assuming that content is entered in this section so it's readily available to copy and paste when uploading into whatever vocabulary tool is being used.  I'll be updating my template according to RVA requirements, but would appreciate your confirmation that contents from this section are transferred manually (and maybe we need to note this somehow in the process document?)
4. In the case of multiple altLabels, how should these be entered?  Comma-separated?? Not an issue with the FSDF Themes vocab, but something we’ll need to be able to do for other entries…

FYI, I've also updated vocab-maintenance.md to clarify a couple of things.  If this wasn't appropriate please re-revise:

(a) Code line 33: added clarification 'Note in this instance "pt" refers to place type and is specific to this example only'
(b) Code line 75: added clarification …'(row 25 in this example)', as it wasn't clear to me what "Data Structure Row" referred to

Cheers,
Jenny  (LongJenny)

**Comment by dr-shorthair** [comment]
@LongJenny can you share the input (Excel) file please? 
Put it in https://github.com/GeoscienceAustralia/FSDF/tree/master/codelists 

(Do you know how to work with Git or GitHub?)

**Comment by LongJenny** [comment]
Nope, sorry, I’m a GitHub newby.  But dragging and dropping seems to have worked and hopefully if I’ve put it in the wrong place or named it incorrectly you can fix that?  Anyway, it’s there now…

**Comment by dr-shorthair** [comment]
OK - I found the following bug:
- the namespace for the members of this vocabulary must end in a / or # 
- you chose the prefix 'URI' - that is a bit unconventional. Probably better to use something related to the content. I've changed it to 'theme'
-
I've tested this with SKOS-Play! and it looks good. 
I've pushed the corrected XLSX over the top of the one you loaded.

**Comment by LongJenny** [comment]
Great, thanks Simon. I’ll have a look at it when I’m back in the office tomorrow

**Comment by dr-shorthair** [comment]
I also changed its name to something much easier – FSDF-themes.xlsx

**Comment by dr-shorthair** [comment]
Thx for the tweaks to the instructions BTW - this is how it should be done!

**Comment by LongJenny** [comment]
@dr-shorthair 

Good pick up on the missing forward slash, and thanks for guidance re naming conventions for URI prefix code as well as filename.

Are you able to give me some feedback on all four of the numbered items I put in my original message in this Issue? e.g. it's still returning odd characters in the place of hyphens and apostrophes...

**Comment by dr-shorthair** [comment]
I've just run SKOS-Play! conversion again and I'm not seeing the problems 1, 3. 

1. `theme:elevation-and-depth` is as expected
2. yes - just copy and paste the output from SKOS-Play! 
3. all the vocabulary metadata is there, associated with the `ConceptScheme`
4. Commas can be used to separate **URIs**, but **not values in a string** by default anyway, as comma is a valid character that could appear in the middle of a string. 

However, you can add another configuration option to force it to split a string into multiple values - see https://skos-play.sparna.fr/play/convert#split 
Another way to get multiple values is to insert additional lines for the additional labels.  All the 'properties' of a URI get associated with just the one entry. 

I've tested the following which illustrates both approaches, and it creates three `skos:altLabel` values. 

![image](https://user-images.githubusercontent.com/608303/121630922-6cee7d80-cac1-11eb-914c-aa964123c41e.png)

**Comment by LongJenny** [comment]
Thanks for your thorough input - much appreciated.  

This is what I'm getting instead of apostrophe (API view from RVA as I find that easier to comprehend): 
![image](https://user-images.githubusercontent.com/74519443/121644068-e6906680-cad5-11eb-876b-b0412430678c.png)

Sounds possible problem is my end only - I'll check on my computer at home and raise it with GA if displaying correctly on my device, but won't update this issue post immediately as I'm off now until next Thursday (loooong weekend!)  If I can't resolve it I may check back with you late next week.  

I hope you enjoy your weekend.

Jenny

**Comment by dr-shorthair** [comment]
Ah – I see. That is something deep in character sets and encoding. If you replace the apostrophe (which is a funny character) with a single quote (which is part of the ASCII character set) then it works (I just tested).

Lucky you for long weekend. We still can’t go further than 25km from home, and didn’t feel like we could plan while our lockdown was in place. We have lost so many holidays in the last 18 months, as we also lost 2020 summer holiday in Kangaroo Island due to the bushfires.
